mpi4jax/collective_ops/allreduce.py

```diff
@@ -133,51 +150,77 @@ def mpi_allreduce_xla_encode_gpu(c, x, token, op, comm):
         _dtype_ptr,
     )
 
-    return _ops.CustomCall(
-        c,
-        b"mpi_allreduce",
-        operands=(
-            x,
-            token,
-        ),
-        shape=sh,
-        opaque=descriptor,
-        has_side_effect=True,
-    )
+    if transpose:
+        if op != _MPI.SUM:
+            raise NotImplementedError(
+                "The linear transpose of Allreduce for {} is not defined".format(op)
+            )
+
+        raise NotImplementedError("TODO implement on GPU transpose of Allreduce")
+    else:
```

Comment:
Wouldn't the same from the cpu version also work here?
If so a disadvantage of having just one primitive is that we have to duplicate this part for every platform.
In https://github.com/PhilipVinc/mpi4jax/pull/35/commits/5a60eaaf8f4c172a9d0d7db4e2be4ef75654be20 I had added it to `xla.translations` directly, because I was assuming it would work.

Reply:
Yes, probably. I simply did not try.